src/client-ocurl/config.ml

```diff
@@ -27,18 +31,53 @@ let pp out self =
     self
   in
   Format.fprintf out
-    "{@[ debug=%B;@ url=%S;@ headers=%a;@ batch_timeout_ms=%d; bg_threads=%d;@ \
-     ticker_thread=%B;@ ticker_interval_ms=%d;@ self_trace=%B @]}"
-    debug url ppheaders headers batch_timeout_ms bg_threads ticker_thread
-    ticker_interval_ms self_trace
+    "{@[ debug=%B;@ url_traces=%S;@ url_metrics=%S;@ url_logs=%S;@ \
+     headers=%a;@ batch_timeout_ms=%d; bg_threads=%d;@ ticker_thread=%B;@ \
+     ticker_interval_ms=%d;@ self_trace=%B @]}"
+    debug url_traces url_metrics url_logs ppheaders headers batch_timeout_ms
+    bg_threads ticker_thread ticker_interval_ms self_trace
 
-let make ?(debug = !debug_) ?(url = get_url ()) ?(headers = get_headers ())
-    ?(batch_timeout_ms = 2_000) ?(bg_threads = 4) ?(ticker_thread = true)
-    ?(ticker_interval_ms = 500) ?(self_trace = false) () : t =
+let make ?(debug = !debug_) ?url ?url_traces ?url_metrics ?url_logs
+    ?(headers = get_headers ()) ?(batch_timeout_ms = 2_000) ?(bg_threads = 4)
+    ?(ticker_thread = true) ?(ticker_interval_ms = 500) ?(self_trace = false) ()
+    : t =
   let bg_threads = max 1 (min bg_threads 32) in
+
+  let url_traces, url_metrics, url_logs =
+    let base_url =
+      match url with
+      | None -> Option.value (get_url_from_env ()) ~default:default_url
+      | Some url -> remove_trailing_slash url
+    in
+    let url_traces =
+      match url_traces with
+      | None ->
+        Option.value
+          (get_url_traces_from_env ())
+          ~default:(base_url ^ "/v1/traces")
+      | Some url -> url
```

Comment:
I'm now wondering if the values coming from the env shouldn't take precedence over the config 🤔 It might be useful, for example, to have a `dev` specific `~config` when developing locally and be able to change it when running the app in production (through env variables). WDYT?

Reply:
Changed in [`e73ea7e` (#56)](https://github.com/imandra-ai/ocaml-opentelemetry/pull/56/commits/e73ea7e6ad9d3610a00f97809085349d2d10b8d6)